### PR TITLE
Start removing externs for `ol/source`

### PR DESCRIPTION
**Pull request by romanzoller**
Start moving externs for `ol/source` (see #7947), progress so far 3 / 26:

 - ~~`Source`~~
   - ~~`Image`~~
     - ~~`ImageArcGISRest`~~
     - `ImageCanvas`
     - `ImageMapGuide`
     - `ImageStatic`
     - `ImageWMS`
     - `Raster`
   - `Tile`
     - `TileDebug`
     - `TileUTFGrid`
     - `UrlTile`
       - `TileImage`
         - `BingMaps`
         - `TileArcGISRest`
         - `TileJSON`
         - `TileWMS`
         - `WMTS`
         - `XYZ`
           - `CartoDB`
           - `OSM`
           - `Stamen`
         - `Zoomify`
       - `VectorTile`
   - `Vector`
     - `Cluster`

### CODE REVIEW

File: src/ol/source/ImageArcGISRest.js [hunk 1]
```diff
@@ -11,6 +11,34 @@ import {assign} from '../obj.js';
...
+ * the remote server.
+ * @property {ol.ImageLoadFunctionType} [imageLoadFunction] Optional function to load an image given
+ * a URL.
+ * @property {Object.<string,*>} params ArcGIS Rest parameters. This field is optional. Service
+ * defaults will be used for any fields not specified. `FORMAT` is `PNG32` by default. `F` is
+ * `IMAGE` by default. `TRANSPARENT` is `true` by default.  `BBOX, `SIZE`, `BBOXSR`, and `IMAGESR`
+ * will be set dynamically. Set `LAYERS` to override the default service layer visibility. See
+ * {@link http://resources.arcgis.com/en/help/arcgis-rest-api/index.html#/Export_Map/02r3000000v7000000/}
+ * for further reference.
+ * @property {ol.ProjectionLike} projection Projection.
```
Comment (fredj) [inline src/ol/source/ImageArcGISRest.js]: can be changed to `module:ol/proj~ProjectionLike`

Comment (romanzoller) [reply src/ol/source/ImageArcGISRest.js]: ok, done

**Comment by ahocevar** [comment]
@romanzoller Are you planning to add more to this pull request, or is it ready for final review and merging?

**Comment by romanzoller** [comment]
Nope, good for review. Sorry, indeed I hadn't communicated that clearly.

@fgravin started to continue where I left off, but I guess that can very well be a new PR, I will ping him to see what the state is.

**Comment by tschaub** [comment]
Thanks @romanzoller!

**Comment by fgravin** [comment]
@ahocevar i'm working on `ol.source`, i'll be finishing what @romanzoller started.

**Pull request merged**